addon/index.js

```diff
@@ -130,28 +149,39 @@ function formatInstallData(install, addon) {
 
 AddonManager.addAddonListener({
   onUninstalling: function(addon) {
-    app.send('addons-uninstalling', {
-      id: addon.id,
-      name: addon.name
-    });
+    if (isIdeatownAddon(addon.id)) {
+      app.send('addon-uninstall:uninstall-started', {
+        id: addon.id,
+        name: addon.name,
+        version: addon.version
+      });
+    }
   },
   onUninstalled: function(addon) {
-    app.send('addons-uninstalled', {
-      id: addon.id,
-      name: addon.name
-    }, addon);
-
-    store.experiments = store.experiments.filter(ex => ex.id !== addon.id);
-    state.updates = require('exclude')(updates, store.experiments);
+    if (isIdeatownAddon(addon.id)) {
+      app.send('addon-uninstall:uninstall-ended', {
+        id: addon.id,
+        name: addon.name,
+        version: addon.version
+      }, addon);
+
+      if (store.experiments[addon.id]) {
+        delete store.experiments[addon.id];
+      }
+    }
   }
 });
 
+function isIdeatownAddon(id) {
+  return Boolean(store.availableExperiments.filter(ae => ae === id).length);
```

Comment:
If I understand correctly, this function just checks if a string is in an array of strings. `indexOf` is a pretty widely-used way to do this:
```
function isIdeatownAddon(id) {
  return store.availableExperiments.indexOf(id) > -1;
}
```